### PR TITLE
feat: add maven flatten plugin

### PR DESCRIPTION
**Pull request by stephaniewang526**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -229,6 +259,14 @@
                   <version>[1.7,)</version>
                 </requireJavaVersion>
                 <requireUpperBoundDeps/>
+                <banDuplicateClasses>
```
Comment (chingor13) [inline pom.xml:262]: Actually, can we separate this into a separate PR so it'll appear as its own entry in the changelog.

Comment (stephaniewang526) [reply pom.xml:262]: SG
